src/openscm_runner/adapters/fair2/fair2.py

```diff
@@ -0,0 +1,316 @@
+
+from scmdata import ScmRun
+from tqdm.autonotebook import tqdm
+
+from ..base import _Adapter
+from ._run_fair import run_fair
+
+from __future__ import division
+
+import datetime as dt
+
+import os
+import numpy as np
+import pandas as pd
+from scipy.interpolate import interp1d
+
+from ..constants import molwt
+
+try:
+    from scmdata import ScmDataFrame, ScmRun
+    has_scmdata = True
+except ImportError:
+    has_scmdata = False
+
+class SSP245WorldEmms:
+    def __init__(self):
+        self._loaded = False
+        self._loaded_fair_history = False
+
+    @property
+    def values(self):
+        if not self._loaded:
+            self._values = (
+                ScmRun(
+                    os.path.join(
+                        os.path.dirname(__file__),
+                        '../SSPs/data/rcmip-emissions-annual-means-4-0-0-ssp-only.csv'),
+                        lowercase_cols=True
+                )
+                .filter(scenario="ssp245", region="World", variable="Emissions*")
+            )
+            self._values = self._values.interpolate([dt.datetime(y, 1, 1) for y in self._values["year"]])
+
+        self._loaded = True
+
+        return self._values
+
+    @property
+    def values_fair_units(self):
+        if not self._loaded_fair_history:
+            ssp_df_hist = self.values
+            for variable in ssp_df_hist.get_unique_meta("variable"):
+                in_unit = ssp_df_hist.filter(variable=variable).get_unique_meta("unit", no_duplicates=True)
+
+                try:
+                    _, fair_unit, context = _get_fair_col_unit_context(variable)
+                except AssertionError:
+                    # FaIR does not model the variable
+                    assert variable in [
+                        'Emissions|F-Gases|HFC|HFC152a',
+                        'Emissions|F-Gases|HFC|HFC236fa',
+                        'Emissions|F-Gases|HFC|HFC365mfc',
+                        'Emissions|F-Gases|NF3',
+                        'Emissions|F-Gases|PFC|C3F8',
+                        'Emissions|F-Gases|PFC|C4F10',
+                        'Emissions|F-Gases|PFC|C5F12',
+                        'Emissions|F-Gases|PFC|C7F16',
+                        'Emissions|F-Gases|PFC|C8F18',
+                        'Emissions|F-Gases|PFC|cC4F8',
+                        'Emissions|F-Gases|SO2F2',
+                        'Emissions|Montreal Gases|CH2Cl2',
+                        'Emissions|Montreal Gases|CHCl3',
+                    ]
+
+                    continue
+
+                if in_unit != fair_unit:
+                    ssp_df_hist = ssp_df_hist.convert_unit(fair_unit, variable=variable, context=context)
+
+            self._values_fair_history = ssp_df_hist
+
+        self._loaded_fair_history = True
+
+        return self._values_fair_history
+
+
+
+# TODO: lazy load this and only load once
+ssp245_world_emms_holder = SSP245WorldEmms()
+
+
+EMISSIONS_SPECIES_UNITS_CONTEXT = pd.DataFrame((
```

Comment:
this should be able to disappear too (again I was hoping the FaIR2.0 pandas interface would be able to handle units conversions)